r1cs/gadgets/std/src/bits/uint64.rs

```diff
@@ -558,4 +615,87 @@ mod test {
             num = num.rotate_right(1);
         }
     }
+
+    #[derive(Copy, Clone, Debug)]
+    enum OperandType {
+        True,
+        False,
+        AllocatedTrue,
+        AllocatedFalse,
+        NegatedAllocatedTrue,
+        NegatedAllocatedFalse,
+    }
+
+    #[test]
+    fn test_uint64_cond_select() {
+        let variants = [
+            OperandType::True,
+            OperandType::False,
+            OperandType::AllocatedTrue,
+            OperandType::AllocatedFalse,
+            OperandType::NegatedAllocatedTrue,
+            OperandType::NegatedAllocatedFalse,
+        ];
+
+        use rand::thread_rng;
+        let rng = &mut thread_rng();
+
+        //random generates a and b numbers and check all the conditions for each couple
+        for _ in 0..1000 {
+            for condition in variants.iter().cloned() {
+                let mut cs = TestConstraintSystem::<Fr>::new();
+                let cond;
+                let a;
+                let b;
+
+                {
+                    let mut dyn_construct = |operand, name| {
+                        let cs = cs.ns(|| name);
+
+                        match operand {
+                            OperandType::True => Boolean::constant(true),
+                            OperandType::False => Boolean::constant(false),
+                            OperandType::AllocatedTrue => {
+                                Boolean::from(AllocatedBit::alloc(cs, || Ok(true)).unwrap())
+                            }
+                            OperandType::AllocatedFalse => {
+                                Boolean::from(AllocatedBit::alloc(cs, || Ok(false)).unwrap())
+                            }
+                            OperandType::NegatedAllocatedTrue => {
+                                Boolean::from(AllocatedBit::alloc(cs, || Ok(true)).unwrap()).not()
+                            }
+                            OperandType::NegatedAllocatedFalse => {
+                                Boolean::from(AllocatedBit::alloc(cs, || Ok(false)).unwrap()).not()
+                            }
+                        }
+                    };
+
+                    cond = dyn_construct(condition, "cond");
+                    a = UInt64::constant(rng.gen());
+                    b = UInt64::constant(rng.gen());
```

Comment:
I opt to do the unit test also (or, in combination)  with non-constant `UInt64`s.

Reply:
Done